app/models/manageiq/providers/ibm_power_hmc/inventory/parser/infra_manager.rb

```diff
@@ -100,6 +102,32 @@ def parse_vm_hardware(vm, lpar)
     )
   end
 
+  def parse_vswitches(host, sys)
+    collector.vswitches[sys.uuid].each do |vswitch|
+      switch = persister.host_virtual_switches.build(
+        :uid_ems => vswitch.uuid,
+        :name    => vswitch.name,
+        :host    => host
+      )
+      persister.host_switches.build(:host => host, :switch => switch)
+    end
+  end
+
+  def parse_vlans(sys)
+    collector.vlans[sys.uuid].each do |vlan|
+      managed_system = persister.hosts.lazy_find(sys.uuid)
+      vswitch = persister.host_virtual_switches.lazy_find(:host => managed_system, :uid_ems => vlan.vswitch_uuid)
+      lan = persister.lans.build(
+        :uid_ems => vlan.uuid,
+        :switch  => vswitch,
+        :tag     => vlan.vlan_id,
+        :name    => vlan.name,
+        :ems_ref => sys.uuid
+      )
+      # persister.host_virtual_lans.build(vlan)  ## No table host_virtual_lans in database ?
```

Comment:
@CallisteH `host_virtual_lans` are an association defined on the core `InfraManager` intended to distinguish from "distributed" lans.  They all go to the same table though.

See https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/infra_manager.rb#L17-L20 and https://github.com/ManageIQ/manageiq/blob/master/app/models/host.rb#L46

If you don't have both distributed and host lans you can just use the `persister.lans` collection like you have above.